lib/soft_delete/soft_deletable.rb

```diff
@@ -78,10 +78,8 @@ def handle_overridden_soft_delete_dependencies
         next unless assn.options[:dependent] == :destroy
 
         # TODO: pass in validate
-        if assn.load_target.respond_to?(:each)
-          assn.load_target.each(&:soft_delete!)
-        else
-          assn.load_target.soft_delete!
+        Array(assn.load_target).each do |target|
+          target.soft_delete! if target.respond_to?(:soft_delete!)
```

Comment:
I think you have revealed an excellent edge case!  But TBH, I am not sure we want to handle it silently like this.  As always, I am open to being convinced otherwise :) but here are my thoughts:

From the client's perspective, they are setting the dependent soft delete behavior.  So their expectation is that the associated record(s) will also be soft deleted. 

And the case we want to handle here is, what if the associated record is not set up for soft delete?  How should we handle that?

What is the client's expectation?  I think it would be surprising for them to find out that the record did not get destroyed/soft deleted.  

Not including the soft deletable behavior in the associated model in that case is likely an oversight.  In that case, I would vote for less surprising behavior and more noisy failure :).  If something isn't set up the way I expect it to be, I want the code to make it super obvious with a BOOM ha ha.

So if my above understanding of the intent is correct, my vote would be to either let the code fail with a standard NoMethodError, or raise a specific SoftDeleteNotImplemented error.

Happy to hear your thoughts!

Reply:
Makes sense. Now it will raise standard NoMethodError if associated model is not implemented with Soft Delete.
Also implemeted skip associated soft delete if required for certain associated models.